### PR TITLE
Feature/updates

### PR DESCRIPTION
**Pull request by Roemer**
I updated the go modules and go itself.
I also added a total memory field.

### CODE REVIEW

File: collector.go [hunk 1]
```diff
@@ -19,13 +19,11 @@ type DockerCollector struct {
 }
 
 func newDockerCollector() *DockerCollector {
-	cli, err := client.NewClientWithOpts(client.FromEnv)
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
```
Comment (0xERR0R) [inline collector.go:22]: 👍

File: collector.go [hunk 2]
```diff
@@ -131,14 +129,21 @@ func (c *DockerCollector) memoryMetrics(ch chan<- prometheus.Metric, containerSt
...
 	ch <- prometheus.MustNewConstMetric(prometheus.NewDesc(
 		"dex_memory_usage_bytes",
 		"Total memory usage bytes",
 		labelCname,
 		nil,
 	), prometheus.CounterValue, float64(memoryUsage), cName)
+	ch <- prometheus.MustNewConstMetric(prometheus.NewDesc(
+		"dex_memory_total_bytes",
+		"Total memory bytes",
+		labelCname,
```
Comment (0xERR0R) [inline collector.go:144]: 👍

**Review by 0xERR0R — APPROVED** [review]
Looks good, thank you!

**Comment by 0xERR0R** [comment]
Thanks for your work! Did you update the grafana dashboard (to show new metric)? If yes, maybe you could share it?

**Comment by Roemer** [comment]
> Thanks for your work! Did you update the grafana dashboard (to show new metric)? If yes, maybe you could share it?

No not yet. I am still playing around with it as I have ~30 containers and it is a bit harder to create one that shows all info and is still usable. But I will gladly share it when I find something that fits my needs.

**Pull request merged**